napari/layers/utils/link_layers.py

```diff
@@ -0,0 +1,145 @@
+        common_attrs = {x for x in common_attrs if not x.startswith("_")}
+    common = common_events & common_attrs - prohibited
+
+    # lastly, discard any method-only events (we just want attrs)
+    for attr in set(common_attrs):
+        # properties do not count as methods and will not be excluded
+        if ismethod(getattr(first_layer.__class__, attr, None)):
+            common.discard(attr)
+
+    return common
+
+
+def experimental_link_layers(
+    layers: Iterable['Layer'], attributes: Iterable[str] = ()
+):
+    """Link ``attributes`` between all layers in ``layers``.
+
+    This essentially performs the following operation:
+
+    .. code-block:: python
+
+       for lay1, lay2 in permutations(layers, 2):
+           for attr in attributes:
+              lay1.events.<attr>.connect(_set_lay2_<attr>)
+
+    Recursion is prevented by checking for value equality prior to setting.
+
+    Parameters
+    ----------
+    layers : Iterable[napari.layers.Layer]
+        The set of layers to link
+    attributes : Iterable[str], optional
+        The set of attributes to link.  If not provided (the default),
+        *all*, event-providing attributes that are common to all ``layers``
+        will be linked.
+
+    Raises
+    ------
+    ValueError
+        If any of the attributes provided are not valid "event-emitting"
+        attributes, or are not shared by all of the layers provided.
+
+    Examples
+    --------
+    >>> data = np.random.rand(3, 64, 64)
+    >>> viewer = napari.view_image(data, channel_axis=0)
+    >>> experimental_link_layers(viewer.layers)
+    """
+    from itertools import permutations, product
+
+    from ...utils.misc import get_equality_operator
+
+    valid_attrs = _get_common_evented_attributes(layers)
+
+    # now, ensure that the attributes requested are valid
+    attr_set = set(attributes)
+    if attributes:
+        extra = attr_set - valid_attrs
+        if extra:
+            raise ValueError(
+                "Cannot link attributes that are not shared by all "
+                f"layers: {extra}. Allowable attrs include:\n{valid_attrs}"
+            )
+    else:
+        # if no attributes are specified, ALL valid attributes are linked.
+        attr_set = valid_attrs
+
+    # now, connect requested attributes between all requested layers.
+    for (lay1, lay2), attr in product(permutations(layers, 2), attr_set):
+
+        key = (id(lay1), id(lay2), attr)
+        if key in _LINKED:
+            continue
+
+        def _make_l2_setter(eq_op, l1=lay1, l2=lay2, attr=attr):
+            # where `eq_op` is an "equality checking function" that is suitable
+            # for the attribute object type.  (see ``get_equality_operator``)
+
+            def setter(event=None):
+                new_val = getattr(l1, attr)
+                # this line is the important part for avoiding recursion
+                if not eq_op(getattr(l2, attr), new_val):
+                    setattr(l2, attr, new_val)
```

Comment:
Once we have the EventedModel wired up which has it's own equality checks will we still need this?

Reply:
I _think_ maybe not?  In any case, whatever solution we come up with there can be directly applied here too, (whether that means that we use the same underling `eq_op` for both things, or that we just don't even need the check here, I'm not sure yet)

Reply:
Yeah that would be great! We'll still need it now for this PR as we're not using the eventedmodel on our layers yet, but I think it would be really nice to have an end state where we've taken care of this stuff there and so it's not needed, but we can see!!